hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -316,6 +322,34 @@ public void persistToken(Token token) {
 		transactionRecordService.includeChangesToToken(token);
 	}
 
+	/**
+	 * Creates a new {@link MerkleToken} and a {@link MerkleEntityId} based on the given new {@link Token}, which was just created.
+	 * Maps the properties between the mutable and immutable token, and later puts the immutable one in state.
+	 * Adds the token's treasury to the known treasuries map.
+	 * @param token - the model of the token to be persisted in state.
+	 *
+	 */
+	public void persistNew(Token token) {
+		/* create new merkle token */
+		final var newMerkleTokenId = token.getId().asMerkle();
+		final var newMerkleToken = new MerkleToken(
+				token.getExpiry(),
+				token.getTotalSupply(),
+				token.getDecimals(),
+				token.getSymbol(),
+				token.getName(),
+				token.isFrozenByDefault(),
+				!token.hasKycKey(),
+				token.getTreasury().getId().asEntityId()
+		);
+		/* map changes */
+		mapModelChangesToMutable(token, newMerkleToken);
+		tokens.get().put(newMerkleTokenId, newMerkleToken);
+		addKnownTreasury(token.getTreasury().getId(), token.getId());
```

Comment:
I think that it would be better if we introduce general steps here in this method. Other than:
1) Create new Instance
2) Populate properties
3) Add new instance
5) Externalise changes

We have step 4) which is updating any auxiliary structures. In the case for `create` -> we are adding `treasury` in the case for update, we are updating the FCOTMRs